code/controllers/subsystem/parallax.dm

```diff
@@ -19,32 +29,76 @@ SUBSYSTEM_DEF(parallax)
 	planet_y_offset = rand(100, 160)
 	planet_x_offset = rand(100, 160)
 
+/datum/controller/subsystem/parallax/Initialize(start_timeofday)
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_MOB_LOGGED_IN, .proc/on_mob_login)
+	throttle_ghost_pop = CONFIG_GET(number/parallax_ghost_disable_pop)
+	throttle_all_pop = CONFIG_GET(number/parallax_disable_pop)
+
 /datum/controller/subsystem/parallax/fire(resumed = 0)
-	if (!resumed)
-		src.currentrun = GLOB.clients.Copy()
-
-	//cache for sanic speed (lists are references anyways)
-	var/list/currentrun = src.currentrun
-
-	while(length(currentrun))
-		var/client/C = currentrun[currentrun.len]
-		currentrun.len--
-		if (!C || !C.eye)
-			if (MC_TICK_CHECK)
-				return
-			continue
-		var/atom/movable/A = C.eye
-		if(!istype(A))
+	//Swap the 2 lists
+	if(!length(currentrun))
+		//Nothing to process here
+		if(!length(queued))
+			return
+		var/temp = currentrun
+		currentrun = queued
+		queued = temp
+		current_run_pointer = 1
+		//Check client count
+		throttle_ghosts = throttle_ghost_pop && length(GLOB.clients) >= throttle_ghost_pop
+		throttle_all = throttle_all_pop && length(GLOB.clients) >= throttle_all_pop
+	//Begin processing the processing queue
+	while(current_run_pointer <= length(currentrun))
+		//Use a pointer, less wasted processing than removing from the list
+		var/client/C = currentrun[current_run_pointer]
+		//Increment the current list pointer, so we process the next element
+		current_run_pointer ++
+		//No client (Disconnected)
+		if(!C)
 			continue
-		for (A; isloc(A.loc) && !isturf(A.loc); A = A.loc);
-
-		if(A != C.movingmob)
-			if(C.movingmob != null)
-				C.movingmob.client_mobs_in_contents -= C.mob
-				UNSETEMPTY(C.movingmob.client_mobs_in_contents)
-			LAZYINITLIST(A.client_mobs_in_contents)
-			A.client_mobs_in_contents += C.mob
-			C.movingmob = A
-		if (MC_TICK_CHECK)
+		C?.parallax_update_queued = FALSE
+		//Do the parallax update (Move it to the correct location)
+		if ((throttle_ghosts && isobserver(C.mob)) || (throttle_all))
+			C?.mob?.hud_used?.freeze_parallax()
+		else
+			C?.mob?.hud_used?.update_parallax()
```

Comment:
You have to be super unlucky for the client to become a null in a synchronous block of code like this (if it's even possible since we don't multi thread).
If there is no sleep between client null check and access to it, you can assume it does exist.
Also, if you wanted to be consistent, you are not checking client in `isosbserver(C.mob)`.

Reply:
Added the null conditional check to C.mob, removed the null check from mob, since it won't be null.